sqldelight-core/src/main/kotlin/com/squareup/sqldelight/core/compiler/MutatorQueryGenerator.kt

```diff
@@ -41,4 +50,86 @@ class MutatorQueryGenerator(private val query: NamedMutator) {
             .build())
         .build()
   }
+
+  /**
+   * The generated mutator type with a single execute() method which performs the query.
+   *
+   * eg:
+   *   private inner class InsertColumns(private val statement: SqlPreparedStatement) {
+   *     fun execute(id: Int): Long {
+   *       statement.bindLong(0, id.toLong())
+   *       val result = statement.execute()
+   *       deferAction {
+   *         (queryWrapper.otherQueries.someSelect + queryWrapper.otherQueries.someOtherSelect)
+   *           .forEach { it.notifyResultSetChanged() }
+   *       }
+   *       return result
+   *     }
+   *   }
+   */
+  fun type(): TypeSpec {
+    // The type of the class:
+    // private inner class InsertColumns
+    val type = TypeSpec.classBuilder(query.name.capitalize())
+        .addModifiers(INNER, PRIVATE)
+
+    val constructor = FunSpec.constructorBuilder()
+
+    // The statement constructor property:
+    // (private val statement: SqlPreparedStatement)
+    constructor.addParameter(STATEMENT_NAME, STATEMENT_TYPE)
+    type.addProperty(PropertySpec.builder(STATEMENT_NAME, STATEMENT_TYPE, PRIVATE)
+        .initializer(STATEMENT_NAME)
+        .build())
+
+    // The execute method:
+    // fun execute(_id: Int): Long
+    val executeMethod = FunSpec.builder(EXECUTE_METHOD)
+        .returns(LONG)
+        .apply {
+          query.arguments.forEachIndexed { index, parameter ->
+            addParameter(parameter.name, parameter.javaType)
+
+            // statement binding code:
+            // statement.bindLong(0, _id.toLong())
+            addCode(parameter.preparedStatementBinder(index))
+          }
+        }
+        .addStatement("val $EXECUTE_RESULT = $STATEMENT_NAME.execute()")
+        .apply {
+          val resultSetsUpdated = mutableListOf<NamedQuery>()
+          query.statement.sqFile().iterateSqliteFiles { psiFile ->
+            if (psiFile !is SqlDelightFile) return@iterateSqliteFiles true
+            resultSetsUpdated.addAll(psiFile.sqliteStatements().namedQueries()
+                .filter { it.tablesObserved.contains(query.tableEffected) })
+            return@iterateSqliteFiles true
+          }
+
+          if (resultSetsUpdated.isEmpty()) return@apply
+
+          // If there are any result sets we update, defer an action to update them:
+          // deferAction {
+          //   (queryWrapper.dataQueries.selectForId + queryWrapper.otherQueries.selectForId)
+          //     .forEach { it.notifyResultSetChanged() }
+          // }
+          // TODO: Only notify queries that were dirtied (check using dirtied method).
+          addCode(CodeBlock.Builder()
+              .add("deferAction {\n")
+              .indent()
+              .add(resultSetsUpdated.map { it.queryProperty }.joinToCode(" + ", "(", ")\n"))
```

Comment:
This should be `addStatement` and include the `.forEach` part after the newline so it automatically gets a continuation indent.